backend/src/main/java/com/soen341/instagram/service/impl/CommentService.java

```diff
@@ -133,4 +153,27 @@ public Comment findComment(final long commentId)
 		}
 		return commentOptional.get();
 	}
+
+	public CommentResponseDTO determineEditable(final CommentResponseDTO commentResponseDTO)
+	{
+		String currentUser = null;
+		if (!(SecurityContextHolder.getContext().getAuthentication() instanceof AnonymousAuthenticationToken))
+		{
+			final Account currentUserRequest = UserAccessor.getCurrentAccount(accountRepository);
+			currentUser = currentUserRequest.getUsername();
+		}
+
+		// if current user matches the comment account or the picture account -> allow
+		// editing
+		if (currentUser != null && (commentResponseDTO.getAccount().equals(currentUser)
+				|| commentResponseDTO.getPictureDTO().getAccount().equals(currentUser)))
```

Comment:
To make this shorter you can save the condition as a boolean and then pass that boolean to setEditable(). For some reason I'm not a fan of using an if else for the purpose of just passing a boolean of the opposite value.

Reply:
Fair point!